### PR TITLE
chore(cards): replace dashboard/connection pf3 cards/layout with pf4

### PR DESCRIPTION
**Pull request by seanforyou23**
This PR upgrades several Card views from PF3 to PF4. I also swapped out any usages of PF3 Card Grid while I was in there. There's a little bit of fine-tuning to do regarding spacing in the main dashboard layout but wanted to go ahead and raise this PR so we can start getting any feedback people may have.

It's an incremental step toward https://github.com/syndesisio/syndesis/issues/6240

Addresses card views in the following components.

- ConnectionCard
- ConnectionsGrid
- ConnectionsGridCell
- ConnectionsMetric
- Dashboard
- IntegrationBoard
- RecentUpdatesCard
- TopIntegrationsCard
- UptimeMetric
- AggregatedMetricCard

Screen Shot 2019-07-30 at 12 51 08 PM

Screen Shot 2019-07-30 at 12 51 32 PM

Screen Shot 2019-07-30 at 12 51 46 PM

### CODE REVIEW

File: app/ui-react/packages/ui/src/Connection/ConnectionsGridCell.tsx [hunk 1]
```diff
@@ -1,12 +1,7 @@
-import { CardGrid } from 'patternfly-react';
 import * as React from 'react';
 
 export class ConnectionsGridCell extends React.Component {
```
Comment (gashcrumb) [inline app/ui-react/packages/ui/src/Connection/ConnectionsGridCell.tsx]: Maybe this one should just be removed then?

File: app/ui-react/packages/ui/src/Connection/ConnectionsGrid.tsx [hunk 2]
```diff
@@ -1,12 +1,8 @@
...
 import * as React from 'react';
 
 export class ConnectionsGrid extends React.Component {
   public render() {
-    return (
-      <CardGrid fluid={true} matchHeight={true}>
-        <CardGrid.Row>{this.props.children}</CardGrid.Row>
-      </CardGrid>
-    );
+    return <Gallery gutter={'sm'}>{this.props.children}</Gallery>;
```
Comment (mcoker) [inline app/ui-react/packages/ui/src/Connection/ConnectionsGrid.tsx]: This creates the empty state as a grid item, too. Can we only include the gallery layout if there are connections?

Screen Shot 2019-07-30 at 5 09 07 PM

Comment (mcoker) [reply app/ui-react/packages/ui/src/Connection/ConnectionsGrid.tsx]: Also looks like you're missing a global variable (`--pf-l-gallery--m-gutter--GridGap`) that the gallery uses to define space responsively. That var comes from the core base CSS. Is it possible you have a newer version of the `Gallery` component than the base CSS?

Screen Shot 2019-07-30 at 5 15 43 PM

File: app/ui-react/packages/ui/src/Dashboard/Dashboard.css [hunk 3]
```diff
@@ -3,19 +3,54 @@
...
+}
+
 .dashboard__connections__actions {
   padding: 0 10px;
 }
 
-.dashboard__metrics .card-pf.card-pf-aggregate-status {
-  padding: 0 10px 24px 10px;
+.dashboard__metrics .pf-l-grid__item {
+  margin-bottom: var(--pf-global--spacer--md);
```
Comment (mcoker) [inline app/ui-react/packages/ui/src/Dashboard/Dashboard.css]: this is from the missing var mentioned in https://github.com/syndesisio/syndesis/pull/6287/files#r308968199. If you pull in that var, you shouldn't need this.

File: app/ui-react/packages/ui/src/Dashboard/Dashboard.css [hunk 4]
```diff
@@ -3,19 +3,54 @@
...
+}
+
+@media screen and (min-width: 768px) {
+  .dashboard__metrics .pf-l-grid__item {
+    margin-bottom: 0;
```
Comment (mcoker) [inline app/ui-react/packages/ui/src/Dashboard/Dashboard.css]: same. ideally we would get rid of this.

File: app/ui-react/packages/ui/src/Connection/ConnectionsGridCell.tsx (hunk 1 above)
Comment (seanforyou23) [reply app/ui-react/packages/ui/src/Connection/ConnectionsGridCell.tsx]: If it's ok, I'd like to make this the very first thing I do in a follow-up PR?

**Comment by seanforyou23** [comment]
Ok just pushed some updates, thx for the feedback. @mcoker I had to manually set [that gutter variable](https://github.com/syndesisio/syndesis/pull/6287/files#diff-4e966155b50ec615c88fe7bb7ec04303R24), not sure why but I think it is to do with the order of stylesheet injection. Let me know if you spot anything obvious, otherwise, I'll work on correcting this in a follow-up PR so that we properly receive the default from core. Lemme know what you think!

**Pull request merged**